### PR TITLE
TF-2694 Clear sort order when mailbox is tapped

### PR DESCRIPTION
**Pull request by tddang-linagora**
### Issue
- #2694 

### Demo

https://github.com/linagora/tmail-flutter/assets/160106668/69395e7d-7a19-4f8b-8dc7-977cfa2304ca

https://github.com/linagora/tmail-flutter/assets/160106668/03b5271f-5b17-4c8a-8615-7082626f362b

### CODE REVIEW

**Review by chibenwa — COMMENTED** [review]
No tests?

File: lib/features/mailbox/presentation/mailbox_controller.dart [hunk 1]
```diff
@@ -550,14 +550,14 @@ class MailboxController extends BaseMailboxController with MailboxActionHandlerM
...
     KeyboardUtils.hideKeyboard(context);
-    mailboxDashBoardController.setSelectedMailbox(presentationMailboxSelected);
     mailboxDashBoardController.clearSelectedEmail();
     if (presentationMailboxSelected.id != mailboxDashBoardController.selectedMailbox.value?.id) {
       mailboxDashBoardController.clearFilterMessageOption();
     }
     _disableAllSearchEmail();
     _replaceBrowserHistory();
     mailboxDashBoardController.closeMailboxMenuDrawer();
+    mailboxDashBoardController.setSelectedMailbox(presentationMailboxSelected);
```
Comment (hoangdat) [inline lib/features/mailbox/presentation/mailbox_controller.dart:560]: Can you write the test to make sure the order of actions in this case? @tddang-linagora

File: test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart [hunk 2]
```diff
@@ -0,0 +1,472 @@
...
+        mailboxId: testMailboxId),
+      propertiesCreated: ThreadConstants.propertiesDefault,
+      propertiesUpdated: ThreadConstants.propertiesUpdatedDefault));
+  });
+
+  test('WHEN user use advanced search, '
+    'THEN user tap on mail box, '
+    'SHOULD reset all advanced search and filter options, '
+    'THEN query get all email with default options',
+  () async {
```
Comment (hoangdat) [inline test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart:420]: how about test case: change sort option in advanced search also?

File: test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart
Comment (hoangdat) [inline test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart:1]: It is `mailbox_dashboard_controller_test`, please structure this file to test.
```
main() {
    group("search") { 
       setUp()
       group("quick search")
       group("advanced search")
       tearDown()
   }
   .....
}
```

**Comment by hoangdat** [comment]
please have the video demo for `search/advanced search`

File: test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart (hunk 2 above)
Comment (tddang-linagora) [reply test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart:420]: The test description is misleading. The second test was covering all cases in advanced search/sort/filter. Will update the test description.

File: test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart
Comment (tddang-linagora) [reply test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart:1]: done

File: test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart (hunk 2 above)
Comment (tddang-linagora) [reply test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart]: Done

**Comment by tddang-linagora** [comment]
> please have the video demo for `search/advanced search`

Done

**Review by hoangdat — APPROVED** [review]

File: test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart [hunk 3]
```diff
@@ -0,0 +1,474 @@
+// Web only, due to the problem only exists in web
```
Comment (dab246) [inline test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart:1]: Should use annotation `@TestOn('${Platform}')`

File: test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart [hunk 4]
```diff
@@ -0,0 +1,474 @@
...
+import 'package:tmail_ui_user/main/bindings/network/binding_tag.dart';
+import 'package:tmail_ui_user/main/utils/email_receive_manager.dart';
+import 'package:tmail_ui_user/features/network_connection/presentation/network_connection_controller.dart'
+    if (dart.library.html) 'package:tmail_ui_user/features/network_connection/presentation/web_network_connection_controller.dart';
+import 'package:uuid/uuid.dart';
+
+import '../../../email/presentation/controller/single_email_controller_test.mocks.dart';
+import 'mailbox_dashboard_controller_test.mocks.dart';
+
+mockControllerCallback() => InternalFinalCallback<void>(callback: () {});
```
Comment (dab246) [inline test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart:100]: What is this used for?

Comment (tddang-linagora) [reply test/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller_test.dart:100]: Mokito has issue with generated mocks when using with Get. The fix is suggested from here:
- https://github.com/dart-lang/mockito/issues/446#issuecomment-896410176

**Review by dab246 — APPROVED** [review]

**Comment by dab246** [comment]
Please rebase it

**Review by dab246 — APPROVED** [review]
lgtm

**Pull request merged**